query-engine/black-box-tests/tests/protocols/mismatched.rs

```diff
@@ -0,0 +1,67 @@
+use crate::helpers::*;
+use query_engine_tests::*;
+
+const JSON_QUERY: &'static str = r###"
```

Comment:
Worth adding the opposite for sending GraphQL to an Engine that expect Json?
(That is the case that happens in PDP Control plane right now when the user has the `jsonProtocol` preview feature activated in their schema)

Reply:
For the sake of completion, I can add it. The code catches an error in parsing, no matter the protocol. So it won't cover a different case. Both the graphql and json queries are wrapped in a json object that would be parsed with serde, so the error might be the same, A test to check that will definitely not harm.